Allow arcgis rest places package to use dev endpoints

MonikaKanduri4525 opened the pull request:
- Added an endpoint param for places method in arcgis-rest-places package which allows to override the built in URL.
- Added test cases 

Pending:
comments for TypeDoc

patrickarlt reviewed (changes requested):
@MonikaKanduri4525 this looks great could you reset the demo URL and work on the typedoc comments based on my example?

I'm also going to ask @gavinr to review this as another contributor.

gavinr-maps commented on packages/arcgis-rest-places/test/findPlacesNearPoint.test.ts:
```diff
@@ -74,4 +74,22 @@ describe("findPlacesNearPoint()", () => {
+      y: 55.9533,
+      radius: 10,
+      endpoint:
+        "https://placesdev-api.arcgis.com/arcgis/rest/services/places-service/v1/places/near-point",
```
Similar to Patrick's comment above, I think all these example URLs in the test spec files should be changed to 
```suggestion
        "https://places-api.arcgis.com/arcgis/rest/services/places-service/v1/places/near-point",
```

gavinr-maps commented on packages/arcgis-rest-places/test/findPlacesNearPoint.test.ts:
```diff
@@ -74,4 +74,22 @@ describe("findPlacesNearPoint()", () => {
+  it("verify endpoint", async () => {
+    fetchMock.mock("*", placeNearPointMockNoMoreResults);
+
+    const response = await findPlacesNearPoint({
```
In these tests files, since we're not using the `response` variable, I think you can leave that off, like this:
```suggestion
    await findPlacesNearPoint({
```

gavinr-maps reviewed (changes requested):
Overall looks good! In addition to Patrick's comments, just a few small things:

MonikaKanduri4525 commented:
@patrickarlt @gavinr-maps - Thanks for the inputs, addressed the feedback here [161a784](https://github.com/Esri/arcgis-rest-js/pull/1170/commits/161a7840e01ba70436ba4245c5c11c53d6a3d362)

patrickarlt reviewed (approved):
LGTM I'll merge once tests pass.
